### ISSUE TITLE
Collapse cascading JSON Schema oneOf findings

### ISSUE THREAD

**Issue by TWiStErRob** [TWiStErRob/net.twisterrob.ghlint#805]
### Version

[GH-Lint 0.5.0](https://github.com/TWiStErRob/net.twisterrob.ghlint/releases/tag/v0.5.0) through [`TWiStErRob/net.twisterrob.ghlint@v0.5.0`](https://github.com/TWiStErRob/net.twisterrob.ghlint/tree/v0.5.0)

### Reproduction steps

1. Create `action.yml`:

   ```yaml
   name: Missing shell
   description: Reproduce cascading oneOf diagnostics
   runs:
     using: composite
     steps:
       - name: Example
         run: echo example
   ```

   The composite run step is invalid because `run` requires `shell`; see the [`github-action.json` step alternatives](https://github.com/SchemaStore/schemastore/blob/195da2efaeb9c5b7aa77f1cd8de25b22a53b13f0/src/schemas/json/github-action.json#L173-L202).

2. Run:

   ```shell
   ghlint --console action.yml
   ```

### Expected behavior

Report one actionable finding for the invalid composite step, ideally explaining that a `run` step requires `shell`.

At minimum, report only the closest schema failure at `/runs/steps/0`, rather than independent findings from unrelated JavaScript and Docker action branches.

### Actual behavior

GH-Lint reports six findings for the single missing property:

```text
Expected any of [[node12, node16, node20, node24]] (/runs/using)
Object does not have some of the required properties [[main]] (/runs)
Value matches against more than one schema. Matched schema indexes [0, 1] (/runs/steps/0)
Expected docker (/runs/using)
Object does not have some of the required properties [[image]] (/runs)
Value does not match against any of the schemas (/runs)
```

Each diagnostic is emitted as an independent SARIF result. Five describe failed JavaScript/Docker alternatives or the aggregate `runs.oneOf`; only the deeper `/runs/steps/0` diagnostic belongs to the selected composite action branch.

### Additional context

[`dev.harrel.jsonschema.Error`](https://javadoc.io/doc/dev.harrel/json-schema/1.6.1/dev/harrel/jsonschema/Error.html) exposes `evaluationPath` and `schemaLocation`, but [`YamlValidation.validate`](https://github.com/TWiStErRob/net.twisterrob.ghlint/blob/7215d3dedfaed06cd9adbbf68bfb694ff75c44fe/modules/ghlint-jsonschema/src/main/kotlin/net/twisterrob/ghlint/yaml/YamlValidation.kt#L46-L50) currently reduces it to only:

```kotlin
YamlValidationProblem(it.error, it.instanceLocation)
```

GH-Lint could preserve the schema/evaluation paths and prune combinator noise:

1. Build the error tree from `evaluationPath`.
2. Partition errors by their nearest combinator ancestor.
3. Treat `oneOf` and `anyOf` children as alternatives, not independent violations.
4. Select the closest alternative using discriminator evidence first, then instance depth and leaf-error count.
5. Keep distinct leaf errors within the selected branch; these can represent genuinely independent violations.
6. Treat `allOf` children as cumulative and retain errors from every component.
7. Suppress a parent combinator error when more actionable child errors are available.
8. Keep raw diagnostics available in verbose/debug output.
9. If no alternative clearly wins, emit one grouped "does not match a supported variant" finding with branch diagnostics as details rather than silently choosing or emitting every branch as separate SARIF results.

Here, `runs.using: composite` is strong discriminator evidence: it fails the JavaScript enum and Docker `const`, while satisfying the composite branch. The composite branch also reaches deeper to `/runs/steps/0`. GH-Lint should therefore discard the JavaScript/Docker branch diagnostics and retain the composite-step failure.

After selecting the composite branch, separate leaf failures should still remain separate findings. For example, a missing `shell` in one step and a disallowed property in another step are independent because they occur at different instance paths within the same selected branch.

A typo can produce multiple mechanically correct errors at the same object, such as unknown `shel` plus missing required `shell`. This could be a later normalization step: correlate `additionalProperties` and `required` failures under the same parent using close property-name edit distance, and emit one suggested correction.

<details>
<summary>Previous real-world example: five findings for <code>workflow_call: null</code></summary>

SchemaStore/schemastore#6270 temporarily constrained `on.workflow_call` to an object. An empty YAML trigger is parsed as `null`, causing GH-Lint to report five findings:

```text
Value is [object] but should be [string] (/on)
Expected any of [[branch_protection_rule, ..., workflow_call, ...]] (/on)
Value is [object] but should be [array] (/on)
Value is [null] but should be [object] (/on/workflow_call)
Value does not match against any of the schemas (/on)
```

SchemaStore/schemastore#6307 fixes the upstream schema regression, but demonstrates how combinator diagnostics become multiple code-scanning alerts across repositories.

</details>

_AI-generated (GPT-5.6 Sol), session: 735ab2c3-5ca6-4a29-815a-614392508b98 in net.twisterrob.ghlint._
